compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -1043,6 +1043,15 @@ impl<'hir> Map<'hir> {
             _ => None,
         }
     }
+
+    pub fn impl_parent(self, did: LocalDefId) -> Option<LocalDefId> {
+        let parent = self.find_parent_node(self.local_def_id_to_hir_id(did))?;
```

Comment:
This function can be implemented as
```suggestion
        let parent = self.tcx.local_parent(did);
        if let DefKind::Impl = self.tcx.def_kind(parent) {
            Some(parent)
        } else {
            None
        }
```

This implementation does not look at the HIR, so is friendlier to incremental compilation.